src/components/Message/MessageLivestream.js

```diff
@@ -1,260 +1,366 @@
-     * @param message A [message object](https://getstream.io/chat/docs/#message_format) to resent.
-     */
-    handleRetry: PropTypes.func,
-    /**
-     * Add or remove reaction on message
-     *
-     * @param type Type of reaction - 'like' | 'love' | 'haha' | 'wow' | 'sad' | 'angry'
-     * @param event Dom event which triggered this function
-     */
-    handleReaction: PropTypes.func,
-    /** If actions such as edit, delete, flag, mute are enabled on message */
-    actionsEnabled: PropTypes.bool,
-    /** DOMRect object for parent MessageList component */
-    messageListRect: PropTypes.object,
-    /**
-     * Handler for actions. Actions in combination with attachments can be used to build [commands](https://getstream.io/chat/docs/#channel_commands).
-     *
-     * @param name {string} Name of action
-     * @param value {string} Value of action
-     * @param event Dom event that triggered this handler
-     */
-    handleAction: PropTypes.func,
-    /**
-     * The handler for hover event on @mention in message
-     *
-     * @param event Dom hover event which triggered handler.
-     * @param user Target user object
-     */
-    onMentionsHoverMessage: PropTypes.func,
-    /**
-     * The handler for click event on @mention in message
-     *
-     * @param event Dom click event which triggered handler.
-     * @param user Target user object
-     */
-    onMentionsClickMessage: PropTypes.func,
-    /**
-     * The handler for click event on the user that posted the message
-     *
-     * @param event Dom click event which triggered handler.
-     */
-    onUserClick: PropTypes.func,
-    /**
-     * The handler for mouseOver event on the user that posted the message
-     *
-     * @param event Dom mouseOver event which triggered handler.
-     */
-    onUserHover: PropTypes.func,
-    /**
-     * The component that will be rendered if the message has been deleted.
-     * All of Message's props are passed into this component.
-     */
-    MessageDeleted: PropTypes.elementType,
-  };
-
-  static defaultProps = {
-    Attachment: DefaultAttachment,
-  };
-
-  state = {
-    actionsBoxOpen: false,
-    reactionSelectorOpen: false,
-  };
-
-  reactionSelectorRef = React.createRef();
-
-  editMessageFormRef = React.createRef();
+// eslint-disable-next-line sonarjs/cognitive-complexity
+const MessageLivestreamComponent = (props) => {
+  const {
+    message,
+    groupStyles,
+    editing,
+    clearEditingState,
+    initialMessage,
+    messageListRect,
+    unsafeHTML,
+    onUserClick: propOnUserClick,
+    handleReaction: propHandleReaction,
+    handleOpenThread: propHandleOpenThread,
+    onUserHover: propOnUserHover,
+    handleRetry: propHandleRetry,
+    handleAction: propHandleAction,
+    updateMessage: propUpdateMessage,
+    onMentionsClickMessage: propOnMentionsClick,
+    onMentionsHoverMessage: propOnMentionsHover,
+    Attachment = DefaultAttachment,
+    t: propT,
+    tDateTimeParser: propTDateTimeParser,
```

Comment:
@jaapbakker88 @vishalnarkhede Vini and I discussed this later, on the next major release we should remove these overrides for translation context